smartapps/initialstate-events/initial-state-event-streamer.src/initial-state-event-streamer.groovy

```diff
@@ -327,37 +325,17 @@ def genericHandler(evt) {
 	eventHandler(key, value)
 }
 
-// This is a handler function for flushing the event buffer
-// after a specified amount of time to reduce the load on ST servers
-def flushBuffer() {
-	def eventBuffer = atomicState.eventBuffer
-	log.trace "About to flush the buffer on schedule"
-	if (eventBuffer != null && eventBuffer.size() > 0) {
-		atomicState.eventBuffer = []
-		tryShipEvents(eventBuffer)
-	}
-}
-
 def eventHandler(name, value) {
 	def epoch = now() / 1000
-	def eventBuffer = atomicState.eventBuffer ?: []
-	eventBuffer << [key: "$name", value: "$value", epoch: "$epoch"]
-
-	if (eventBuffer.size() >= 10) {
-		// Clear eventBuffer right away since we've already pulled it off of atomicState to reduce the risk of missing
-		// events.  This assumes the grokerSubdomain, accessKey, and bucketKey are set correctly to avoid the eventBuffer
-		// from growing unbounded.
-		atomicState.eventBuffer = []
-		tryShipEvents(eventBuffer)
-	} else {
-		// Make sure we persist the updated eventBuffer with the new event added back to atomicState
-		atomicState.eventBuffer = eventBuffer
-	}
-	log.debug "Event added to buffer: " + eventBuffer
+
+	def event = new JsonSlurper().parseText("{\"key\": \"$name\", \"value\": \"$value\", \"epoch\": \"$epoch\"}")
+
+	tryShipEvents(event)
+	
+	log.debug "Shipped Event: " + event
 }
 
-// a helper function for shipping the atomicState.eventBuffer to Initial State
-def tryShipEvents(eventBuffer) {
+def tryShipEvents(event) {
```

Comment:
@kris-schaller Yes, this is correct, it will make an HTTP request for every event that the SmartApp is configured to listen to. This is opposed to the previous version that was using an array on atomicState in order to store un-shipped events in a buffer/queue. The buffering (batching) was removed due to issues with the atomicState for storing and retrieving event arrays as well as issues with scheduled tasks to flush buffers that aren't full but are aging. These issues made the SmartApp unusable for most if not all of it's users as of late.